.gitlab/common/macos.yml

```diff
@@ -35,8 +35,9 @@
     echo "Temporary folder created, TMPDIR=$TMPDIR -> $NEWTMPDIR"
 
 .install_python_dependencies:
-  # Create custom temporary folder to isolate jobs from each other
-  # We have to symlink it to /tmp/gitlabci to avoid some path length issues (sockets should be <= 104 characters on MacOS)
+  - export PATH="$(pyenv root)/bin:$PATH"
+  - eval "$(pyenv init -)"
+  - export PYENV_VERSION=datadog-agent-python-$PYTHON_VERSION
```

Comment:
Is PYTHON_VERSION guaranteed to be set? I got the impression from my previous attempts that it wasn't (e.g. [this run](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/1075691895#L73)), and there was a bunch of logic [originally](https://github.com/DataDog/datadog-agent/pull/39719/files#diff-fc56381cbf54fdcd75b13d20f04f9aad90b92d420bb3f8993e647cd681d8c02eL49) to deal with its absence.

I've seen the pipeline pass, so maybe PYENV_VERSION is not affecting anything?

Reply:
Amended with precondition checks, let's see.